### PR TITLE
[FDS-2218] Update the README

### PR DESCRIPTION
**Pull request by jaymedina**
## problem
The `README.md` needed to be updated for clarity in certain parts, particularly the installation instructions and use-case examples.

## solution
- [x] Overall re-structure and stylistic changes
- [x] Add a `TL;DR` section to provide a quick summary of `schematic` and its uses
- [x] Improve developer setup and contribution instructions

Items from Jira tickets:
- [x] Update development environment instructions for a wider audience: [ticket](https://sagebionetworks.jira.com/browse/FDS-1172) 
- [x] Consolidate installation instructions between Confluence and GitHub: [ticket](https://sagebionetworks.jira.com/browse/FDS-2476)
- [x] Point readers to service desk for help: [ticket](https://sagebionetworks.jira.com/browse/FDS-470)
- [x] Fix hyperlink in **Testing** section: [ticket](https://sagebionetworks.jira.com/browse/FDS-2467)
- [x] Fix CLI reference link: [ticket](https://sagebionetworks.jira.com/browse/FDS-2449)

## testing
N/A

### CODE REVIEW

File: README.md
Comment (BryanFauble) [inline README.md:7]: Small tip. I use this vscode extension to automatically create/update the table of contents on save of the file:
https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one

You might find it useful as you're likely adding more sections.

Comment (jaymedina) [reply README.md:7]: nifty. thanks!

File: README.md
Comment (jaymedina) [inline README.md:230]: This caveat is mentioned because of [this logic](https://github.com/Sage-Bionetworks/schematic/blob/90930cd876e88970ddea0ee1c0e28df371d823f2/schematic/store/synapse.py#L380) in the code.

To elaborate: the [`synapse_configuration_path`](https://github.com/Sage-Bionetworks/schematic/blob/1e39a05759fbeb7644d1e4582723beee49c16e33/schematic/configuration/configuration.py#L102) method always assumes [`_parent_directory`](https://github.com/Sage-Bionetworks/schematic/blob/1e39a05759fbeb7644d1e4582723beee49c16e33/schematic/configuration/configuration.py#L50) is the current working directory. This tells the python client to look for the `.synapseConfig` in whatever directory you're currently on, unless configured otherwise.

In this example I am repeating the logic [here](https://github.com/Sage-Bionetworks/schematic/blob/90930cd876e88970ddea0ee1c0e28df371d823f2/schematic/store/synapse.py#L380)
```
>>> import synapseclient
>>> syn = synapseclient.Synapse(configPath=os.getcwd(), cache_root_dir=os.getcwd())
>>> syn.login()

UPGRADE AVAILABLE

A more recent version of the Synapse Client (4.5.1) is available. Your version (4.3.1) can be upgraded by typing:
    pip install --upgrade synapseclient

Python Synapse Client version 4.5.1 release notes

https://python-docs.synapse.org/news/

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.venv/lib/python3.9/site-packages/synapseclient/client.py", line 753, in login
    raise SynapseNoCredentialsError("No credentials provided.")
synapseclient.core.exceptions.SynapseNoCredentialsError: No credentials provided.
```

And in this example, I remove the `configPath` and `cache_root_dir` which tells SynPy to assume `.synapseConfig` and `.synapseCache` are in the home directory, which is where they are created by default. This allowed me to log into Synapse despite not being in home:

```
>>> syn = synapseclient.Synapse()
>>> syn.login()

UPGRADE AVAILABLE

A more recent version of the Synapse Client (4.5.1) is available. Your version (4.3.1) can be upgraded by typing:
    pip install --upgrade synapseclient

Python Synapse Client version 4.5.1 release notes

https://python-docs.synapse.org/news/

Welcome, Jenny Medina!

INFO:synapseclient_default:Welcome, Jenny Medina!

>>> 
```

File: README.md
Comment (BryanFauble) [inline README.md:522]: Should this information live in the `CONTRIBUTION.md` file instead?

**Review by BryanFauble — APPROVED** [review]
This is great! Amazing usage of formatting using the note, important, and warning block!

**Review by andrewelamb — APPROVED** [review]

File: README.md
Comment (jaymedina) [reply README.md]: Hi @BryanFauble good point - I've consolidated this section with what's in `CONTRIBUTION.md` and moved everything other in[ this commit](https://github.com/Sage-Bionetworks/schematic/pull/1510/commits/c3d45527dbc91374f14b31f710021120f9512cfa).

This is ready for a final review. Thanks!

File: CONTRIBUTION.md
Comment (BryanFauble) [inline CONTRIBUTION.md:62]: Is this the correct link? It leads me to a 404 not found

**Review by BryanFauble — APPROVED** [review]
1 comment on a dead link, otherwise LGTM!

File: CONTRIBUTION.md
Comment (GiaJordan) [inline CONTRIBUTION.md:48]: Should we include a reminder to run pre-commit manually on all the files if the hooks weren't enabled?

File: CONTRIBUTION.md
Comment (GiaJordan) [inline CONTRIBUTION.md:51]: Typically in this repo we create a merge commit when we merge PRs. @andrewelamb @linglp did we want this to be different for contributors?

File: CONTRIBUTION.md
Comment (GiaJordan) [inline CONTRIBUTION.md:78]: @jaymedina I see you didn't make any changes to this section so this comment isn't for you it's just to note that this section is out of date and will need to be updated in the future

**Comment by GiaJordan** [comment]
@jaymedina thank you so much for your effort on this! I just left a couple of small comments/questions.

File: CONTRIBUTION.md
Comment (thomasyu888) [reply CONTRIBUTION.md:51]: I recommend sticking with original operations and not squashing and you can decide as a group if you want to squash commits

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!

File: README.md
Comment (jaymedina) [reply README.md:165]: Right now the codebase looks for the environment var `SYNAPSE_ACCESS_TOKEN` https://github.com/Sage-Bionetworks/schematic/blob/49c701c1edef7e67eea1ae21216ab9e168c137e5/schematic/store/synapse.py#L369 

I think changing it to `SYNAPSE_AUTH_TOKEN` would lead to access issues on the users' side

File: CONTRIBUTION.md
Comment (jaymedina) [reply CONTRIBUTION.md:79]: Sounds good! Did you want me to make a Jira ticket for this so we have it in our radar?

Comment (jaymedina) [reply CONTRIBUTION.md:79]: I made a Jira ticket just in case! https://sagebionetworks.jira.com/browse/FDS-2516

Comment (GiaJordan) [reply CONTRIBUTION.md:79]: @jaymedina thank you! I thought I had done so but it looks like I hadn't. Thanks for creating!

Comment (jaymedina) [reply CONTRIBUTION.md:79]: No worries! It looks like there's a ruleset in this repo that requires your approval before merge since there have been changes requested. I'm going to re-request a review. Thanks in advance!

**Comment by GiaJordan** [comment]
@jaymedina thanks for making those changes! 
When I was looking through the `README` file though it looks like the section links from `Contribution Guidelines` through `Reporting Bugs or feature requests` don't lead anywhere now?

File: CONTRIBUTION.md
Comment (GiaJordan) [reply CONTRIBUTION.md:79]: @jaymedina yep no problem!

**Comment by jaymedina** [comment]
@GiaJordan you're right, that's because I moved those sections out of the `README` and into `CONTRIBUTING` - I'll go ahead and remove them from the table of contents. Thanks for the catch!

**Comment by GiaJordan** [comment]
@jaymedina that makes sense, thanks!

**Review by GiaJordan — APPROVED** [review]
Thanks again @jaymedina ! approving before tests finish as changes are constrained to the md doc files

**Pull request merged**
